python/lsst/ts/observatory/control/utils/extras/guider_roi.py

```diff
@@ -0,0 +1,792 @@
+            Angle from boresight in degrees
+
+        Returns
+        -------
+        deltam : float
+            Change in magnitude due to vignetting
+        """
+        vigfraction = self.vigspline(angle)
+        deltam = -2.5 * np.log10(vigfraction)
+        return deltam
+
+    def _get_catalog_data_from_butler(self, hp_indices: list[int]) -> list[Table]:
+        """Get catalog data from butler using HEALPix dimensions."""
+        tables: list[Table] = []
+
+        if not DM_STACK_AVAILABLE:
+            self.log.warning(
+                "DM stack (including Butler) not available for catalog data access"
+            )
+            return tables
+
+        if self.butler is None:
+            return tables
+
+        try:
+            # Query each HEALPix pixel individually
+            for hp_idx in hp_indices:
+                try:
+                    # Create data ID for HEALPix dimension
+                    dataId = {self.healpix_dim: int(hp_idx)}
+
+                    # Query datasets for this specific HEALPix pixel
+                    datasets = list(
+                        self.butler.registry.queryDatasets(
+                            self.catalog_name,
+                            collections=[self.collection],
+                            where=f"{self.healpix_dim} = pixel_id",
+                            bind={"pixel_id": int(hp_idx)},
+                        )
+                    )
+
+                    if datasets:
+                        # Get the catalog for this pixel
+                        catalog = self.butler.get(datasets[0])
+                        if len(catalog) > 0:
+                            tables.append(catalog)
+                    else:
+                        # Try direct get approach as fallback
+                        try:
+                            catalog = self.butler.get(
+                                self.catalog_name,
+                                dataId=dataId,
+                                collections=self.collection,
+                            )
+                            if len(catalog) > 0:
+                                tables.append(catalog)
+                        except (DatasetNotFoundError, Exception):
+                            self.log.warning(
+                                f"No catalog data found for HEALPix pixel {hp_idx}"
+                            )
+
+                except Exception as e:
+                    self.log.warning(
+                        f"Could not load catalog for HEALPix {hp_idx}: {e}"
+                    )
+                    continue
+
+        except Exception as e:
+            self.log.warning(f"Could not load catalog data from butler: {e}")
+
+        return tables
+
+    def get_guider_rois(
+        self,
+        ra: float,
+        dec: float,
+        sky_angle: float,
+        roi_size: int,
+        roi_time: int,
+        band: str,
+        npix_edge: int = 50,
+        npix_boundary: int = 25,
```

Comment:
@aaronroodman I wonder if you could look at npix_boundary, it is an argument to get_guider_rois but is never used.

In case it helps there is this comment in the code about line 644

            # currently just have star's central locations
            # find the location of the ROI containing this star, such that
            # the star is not within npix_boundary of the edge of the ROI
            # and the ROI does not cross the midline break
            
            but it is never used.

Reply:
npix_boundary can be removed.  I had thought I needed two boundaries, but ended up just used the npix_edge one